module/src/clients/surface_temp.c

```diff
@@ -118,8 +135,19 @@ static int ssam_temp_hwmon_read_string(struct device *dev,
 	if (attr != hwmon_temp_label)
 		return -EOPNOTSUPP;
 
-	// TODO: get label from SAM
-	*str = "<not yet implemented>";
+	// Get the label from the SSAM if we haven't retrieved it yet.
```

Comment:
Maybe we should do this upfront when the driver loads? Could simplify things a bit.

Reply:
Hmm, yeah, okay, that's definitely cleaner... we do that now.